File: Lists/src/main/java/Node.java
/**
 * This class is node for single linked list.
 */
public class Node {
  Node next = null;
  Car data;
  public Node(Car data, Node next) {
    this.next = next;
    this.data = data;
  }
  public Node getNext() {
    return next;
  }
  public Car getData() {
    return data;
  }
  public void setNext(Node next) {
    this.next = next;
  }
  public void setData(Car data) {
    this.data = data;
  }
}

File: Test_3.Task_5/Test_3.Task_5/Node.cs
namespace Test_3.Task_5
{
  class Node<Type>
  {
    private Type value;
    private Node<Type> next;

    public Node(Type value)
    {
      this.value = value;
      next = this;
    }

    public Node()
    {
      next = this;
    }

    public Type Value { get { return value; } set { this.value = value; } }
    public Node<Type> Next { get { return next; } set { next = value; } }
  }
}
